scripts/Test-LCOW-UVM.ps1

```diff
@@ -0,0 +1,149 @@
+# benchstat via  `go install golang.org/x/perf/cmd/benchstat@latest`
+
+[CmdletBinding()]
+param (
+    [ValidateSet('Test', 'Bench', 'List', 'Shell')]
+    [alias('a')]
+    [string]
+    $Action = 'Bench',
+
+    [string]
+    $Note = '',
+
+    # test parameters
+    [int]
+    $Count = 1,
+
+    [string]
+    $BenchTime = '5s',
+
+    [string]
+    $Timeout = '10m',
+
+    [alias('tv')]
+    [switch]
+    $TestVerbose,
+
+    [string]
+    $Run = '',
+
+    [string]
+    $CodePath = '.',
+
+    [string]
+    $OutDirectory = '.\test\results',
+
+    # uvm parameters
+
+    [string]
+    $BootFilesPath = 'C:\ContainerPlat\LinuxBootFiles',
+
+    [ValidateSet('vhd', 'initrd')]
+    [string]
+    $BootFSType = 'vhd',
+
+    [switch]
+    $DisableTimeSync,
+
+    # gcs test/container options
+
+    [string]
+    $ContainerRootFSMount = '/run/rootfs',
+
+    [string]
+    $ContainerRootFSPath = (Join-Path $BootFilesPath 'rootfs.vhd'),
+
+    [string]
+    $GCSTestMount = '/run/bin',
+
+    [string]
+    $GCSTestPath = '.\bin\test\gcstest',
+
+    [switch]
+    $MountGCSTest,
+
+    [string]
+    $Feature = ''
+)
+
+Import-Module ( Join-Path $PSScriptRoot Testing.psm1 ) -Force
+
+$CodePath = Resolve-Path $CodePath
+$OutDirectory = Resolve-Path $OutDirectory
+$BootFilesPath = Resolve-Path $BootFilesPath
+$ContainerRootFSPath = Resolve-Path $ContainerRootFSPath
+$GCSTestPath = Resolve-Path $GCSTestPath
+
+$shell = ( $Action -eq 'Shell' )
+
+if ( $shell ) {
+    $cmd = 'ash'
+} else {
+    $date = Get-Date
+    $waitfiles = "$ContainerRootFSMount"
+    $gcspath = 'gcstest'
+    if ( $MountGCSTest ) {
+        $waitfiles += ",$GCSTestMount"
+        $gcspath = "$GCSTestMount/gcstest"
+    }
+
+    $pre = "wait-paths -p $waitfiles -t 5 ; " + `
+        'echo nproc: `$(nproc) ; ' + `
+        'echo kernel: `$(uname -a) ; ' + `
+        'echo gcs.commit: `$(cat /info/gcs.commit 2>/dev/null) ; ' + `
+        'echo gcs.branch: `$(cat /info/gcs.branch 2>/dev/null) ; ' + `
+        'echo tar.date:   `$(cat /info/tar.date 2>/dev/null) ; ' + `
+        'echo image.name: `$(cat /info/image.name 2>/dev/null) ; ' + `
+        'echo build.date: `$(cat /info/build.date 2>/dev/null) ; '
+
+    $testcmd, $out = New-TestCommand `
+        -Action $Action `
+        -Path $gcspath `
+        -Name gcstest `
+        -OutDirectory $OutDirectory `
+        -Date $date `
+        -Note $Note `
+        -TestVerbose:$TestVerbose `
+        -Count $Count `
+        -BenchTime $BenchTime `
+        -Timeout $Timeout `
+        -Run $Run `
+        -Feature $Feature `
+        -Verbose:$Verbose
+
+    $testcmd += " `'-rootfs-path=$ContainerRootFSMount`' "
+    $cmd = $pre + $testcmd
+}
+
+$boot = '.\bin\tools\uvmboot.exe -gcs lcow ' + `
+    '-fwd-stdout -fwd-stderr -output-handling stdout ' + `
+    "-boot-files-path $BootFilesPath " + `
+    "-root-fs-type $BootFSType " + `
+    '-kernel-file vmlinux ' + `
```

Comment:
wouldn't this depend on the bootfs type?

Reply:
I dont think so?
I just tried with uvm boot, and I can boot from `initrd` and `vhd` with both `kernel` and `vmlinux`